src/app/services/wallet.service.ts

```diff
@@ -121,73 +164,106 @@ export class WalletService {
       .post(this.coreUrl, {
         jsonrpc: "2.0",
         id: "jsonrpc",
-        method: "wallet_list"
+        method: "wallet_list",
       })
       .subscribe(
-        data => {
-          data["result"].forEach(wallet => {
+        (data) => {
+          data["result"].forEach((wallet) => {
             walletListFromClient.push({ id: wallet });
           });
           this.walletList.next(walletListFromClient);
         },
-        error => {
-          console.log("Error", error);
+        (error) => {
+          alert(`syncWalletList error ${error}`);
         }
       );
   }
 
-  checkWalletBalance(walletId: string, passphrase: string): Observable<string> {
+  checkWalletBalance(
+    walletId: string,
+    passphrase: string,
+    enckey: string
+  ): Observable<string> {
     return this.http.post<string>(this.coreUrl, {
       jsonrpc: "2.0",
       id: "jsonrpc",
       method: "wallet_balance",
       params: [
         {
           name: walletId,
-          passphrase: _.isNil(passphrase) ? "" : passphrase
-        }
-      ]
+          passphrase: _.isNil(passphrase) ? "" : passphrase,
+          enckey: _.isNil(passphrase) ? "" : enckey,
+        },
+      ],
     });
   }
-  checkWalletAddress(walletId: string, passphrase: string): Observable<string> {
+  checkWalletAddress(
+    walletId: string,
+    passphrase: string,
+    enckey: string
+  ): Observable<string> {
     return this.http.post<string>(this.coreUrl, {
       jsonrpc: "2.0",
       id: "jsonrpc",
       method: "wallet_listTransferAddresses",
       params: [
         {
           name: walletId,
-          passphrase: _.isNil(passphrase) ? "" : passphrase
-        }
-      ]
+          passphrase: _.isNil(passphrase) ? "" : passphrase,
+          enckey: _.isNil(passphrase) ? "" : enckey,
+        },
+      ],
     });
   }
 
-  checkWalletViewKey(walletId: string, passphrase: string): Observable<string> {
+  checkWalletEncKey(walletId: string, passphrase: string): Observable<string> {
```

Comment:
I think we don't need to use Observable here, Observable is more a event stream. Here it is more like an API , I think we can simply use `axios` and return a Promise here. But Observable then toPromise is also fine right now.

Reply:
ok, i'll change